src/docfx/lib/HtmlUtility.cs

```diff
@@ -165,24 +167,27 @@ public static string TransformXref(string html, Func<string, bool, int, (string?
                     continue;
                 }
 
-                var xref = HttpUtility.HtmlDecode(node.GetAttributeValue("href", ""));
+                var rawSource = HttpUtility.HtmlDecode(node.GetAttributeValue("data-raw-html", null) ?? node.GetAttributeValue("data-raw-source", null));
+                var isShorthand = rawSource?.StartsWith("@") ?? false;
 
-                var raw = HttpUtility.HtmlDecode(
-                    node.GetAttributeValue("data-raw-html", null) ?? node.GetAttributeValue("data-raw-source", null) ?? $"<span class=\"xref\">{HttpUtility.HtmlEncode(UrlUtility.SplitUrl(xref).path)}</span>");
+                var href = HttpUtility.HtmlDecode(node.GetAttributeValue("href", null));
+                var uid = href == null ? HttpUtility.HtmlDecode(node.GetAttributeValue("uid", null)) : default;
 
-                var isShorthand = raw.StartsWith("@");
-
-                var (resolvedHref, display) = transform(xref, isShorthand, columnOffset);
+                var (resolvedHref, display) = resolveXref(
+                    href == null ? null : (SourceInfo<string>?)new SourceInfo<string>(href, block?.ToSourceInfo(columnOffset: columnOffset)),
```

Comment:
since `href` is nullable, can just be `(SourceInfo<string>?)new SourceInfo<string>(href, block?.ToSourceInfo(columnOffset: columnOffset)`

Reply:
This will make this parameter always not null.

are you suggesting change the [GetXref](https://github.com/dotnet/docfx/pull/5753/files#diff-1d01eb080be0edc3ee154a3655a2df8fR186)'s parameter type?
from `Source<string>?` to `Source<string?>`, and do the check there?

Reply:
Still let call-site decide what to do when href/uid is null.